src/Structure/Content/Shortcode.php

```diff
@@ -35,7 +51,27 @@ abstract class Shortcode implements ShortcodeInterface {
 	public function register() {
 		$this->validate_tag();
 
-		add_shortcode( $this->tag, [ $this, 'render' ] );
+		add_shortcode( $this->tag, [ $this, 'process_output' ] );
+	}
+
+	/**
+	 * Process the shortcode output.
+	 *
+	 * This method is an intermediary between the shortcode callback and the render method that engineers
+	 * are required to implement as part of the shortcode contract. The main purpose of this process is to apply
+	 * attributes and content to the Shortcode class and make them available at the time `render` is called.
+	 *
+	 * @param array  $attributes The shortcode attributes.
+	 * @param string $content    The shortcode content.
+	 *
+	 * @author Jeremy Ward <jeremy.ward@webdevstudios.com>
+	 * @since  2020-01-31
+	 */
+	public function process_output( array $attributes = [], string $content = '' ) {
```

Comment:
In my tests, I'm showing `$attributes` is an empty string (``) when no attributes are defined, which is causing a failure with typing `$attributes` as an array. I think we need to drop the typehinting on `$attributes`. 😞 

I suspect this is the error Rebekah highlighted in the slack conversation.

Reply:
Same, fatal error now when not passing any attributes.

Reply:
I've removed the typehint and default assignment per feedback from y'all. I've left the assignment for `$content` since that doesn't seem to have any bearing.